update README.md on how to install toolchain and components (#6294)
rust-lang/rustfmt: README.md

To run rustfmt after this, use `cargo run --bin rustfmt -- filename`. See the
notes above on running rustfmt.

> All command should be run in the project's root directory.
> 

Reviewer: I don't think this is correct. You should be able to install the nightly toolchain and components from anywhere.

```suggestion
```

Author: I have completly re-written the section.

### Update Toolchain

> Make sure to have installed the correct toolchain and all components.
> The needed parts can be found in the file `rust-toolchain`. For example:

Reviewer: ```suggestion
rustfmt depends on compiler internals and therefore pins a specific version of the nightly compiler.
Make sure you've installed the correct nightly [toolchain](https://rust-lang.github.io/rustup/concepts/toolchains.html) and all [components](https://rust-lang.github.io/rustup/concepts/components.html) listed in rustfmt's [`rust-toolchain` file](https://github.com/rust-lang/rustfmt/blob/master/rust-toolchain).
```

It's probably worth mentioning this information before telling users to build rustfmt with `cargo build`

```
[toolchain]
channel = "nightly-2024-08-17"
components = ["llvm-tools", "rustc-dev"]

> ```

Reviewer: I'd prefer not to list the content of the `rust-toolchain` file inline. Instead, I'd rather linked out to the file ``[`rust-toolchain`](https://github.com/rust-lang/rustfmt/blob/master/rust-toolchain)``, and let users know that they need to install the exact nightly version of the compiler that rustfmt has pinned in its `rust-toolchain` file.

According to this content, install the toolchain with `rustup install`:

> ```
> rustup install nightly-2024-08-17
> ```

Reviewer: Lets point users to the `rust-toolchain` file to get the exact date:

````suggestion
```
rustup install nightly-YYYY-MM-DD
```
````
